### DISCUSSION TITLE
How can I add borders to the completions and completions docs

### DISCUSSION THREAD

**Discussion by batusai513** [folke/noice.nvim#452] [Q&A] [answered]
Hi, first of all, awesome job making this plugin and all the efforts invested to the neovim ecosystem .

I wanted to ask how to add borders to the completions and completions documentations, I'm using LazyVim with the following override:
```lua
  {
    "folke/noice.nvim",
    opts = {
      popupmenu = {
        backend = "nui",
      },
      presets = {
        lsp_doc_border = true,
      },
      ---@type NoiceConfigViews
      views = {
        popupmenu = {
          border = {
            style = "rounded",
          },
        },
      },
      cmdline = {
        view = "cmdline",
        format = {
          search_down = {
            view = "cmdline",
          },
          search_up = {
            view = "cmdline",
          },
        },
      },
    },
  },
  ```
<img width="1509" alt="Screenshot 2023-05-03 at 09 56 36" src="https://user-images.githubusercontent.com/996671/235861368-7b5bd16c-0b23-45ca-8587-e2062a55e0cf.png">

I also wanted to ask for an overview on which view belongs to each functionality, and how the configs interact with cmp, I was trying to check in from the code but I got lost.

thanks.

**Comment by yutkat** [comment, marked as answer]
I think this is not noice.nvim setting, but an auto-completion(nvim-cmp?) setting.

**Reply by batusai513** [reply]
You are right, by changing the setting in cmp plugin to add borders it works, and that is the part that is a bit confusing for me, what are the boundaries for noice and cmp, from the docs I see that there `backend = "cmp",` for `popupmenu` and given that, you can configure the view with from within noice.

**Reply by yutkat** [reply]
It means that you can use cmp-cmdline instead of noice for completion popups on cmdline. cmp and noice.nvim are quite clearly separated.

**Reply by batusai513** [reply]
ok, I didn't understand at first that cmp was only being used on popupmenus on cmdline if configured to do so, thanks for the clarification.
